devicetypes/smartthings/zwave-temp-humidity-sensor.src/zwave-temp-humidity-sensor.groovy

```diff
@@ -0,0 +1,213 @@
+}
+
+def configure() {
+	response([
+		secure(zwave.batteryV1.batteryGet()),
+		"delay 500",
+		secure(zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 0x05)), // humidity
+		"delay 500",
+		secure(zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 0x01)), // temperature
+		"delay 500",
+		secure(zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 0x0B)), // dew point
+		"delay 500",
+		secure(zwave.wakeUpV2.wakeUpNoMoreInformation())
+	])
+}
+
+def parse(String description) {
+	def results = []
+
+	if (description.startsWith("Err")) {
+		results += createEvent(descriptionText: description, displayed: true)
+	} else {
+		def cmd = zwave.parse(description)
+		if (cmd) {
+			results += zwaveEvent(cmd)
+		}
+	}
+
+	log.debug "parse() result ${results.inspect()}"
+
+	return results
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensorbinaryv1.SensorBinaryReport cmd) {
+	// ignore, to prevent override of SensorBinaryReport
+	[]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	log.debug "Event: ${cmd.event}, Notification type: ${cmd.notificationType}"
+	def result = []
+	def value
+	def description
+
+	if (cmd.notificationType == 0x10) {  // Mold Environment Detection
+		switch (cmd.event) {
+			case 0x00:
+				value = "clear"
+				description = "Mold environment not detected"
+				break
+			case 0x02:
+				value = "moldDetected"
+				description = "Mold environment detected"
+				break
+			default:
+				log.debug "Not handled event type: ${cmd.event}"
+				break
+		}
+
+	result = createEvent(name: "isMoldEnvironment", value: value, descriptionText: description, isStateChange: true, displayed: true)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	def map = [name: "battery", unit: "%", isStateChange: true]
+	state.lastbatt = now()
+
+	if (cmd.batteryLevel == 0xFF) {
+		map.value = 1
+		map.descriptionText = "$device.displayName battery is low!"
+	} else {
+		map.value = cmd.batteryLevel
+	}
+
+	createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	def results = []
+	def map = [:]
+
+	switch (cmd.sensorType) {
+		case 0x01:
+			map.name = "temperature"
+			map.unit = temperatureScale
+			map.value = convertTemperatureIfNeeded(cmd.scaledSensorValue, cmd.scale == 1 ? "F" : "C", cmd.precision)
+			map.displayed = true
+			map.isStateChange = true
+			break
+		case 0x05:
+			map.name = "humidity"
+			map.value = cmd.scaledSensorValue.toInteger()
+			map.unit = "%"
+			map.displayed = true
+			map.isStateChange = true
+			break
+		case 0x0B:
+			map.name = "dewPoint"
+			map.unit = temperatureScale
+			map.value = convertTemperatureIfNeeded(cmd.scaledSensorValue, cmd.scale == 1 ? "F" : "C", cmd.precision)
+			map.displayed = true
+			map.isStateChange = true
+		default:
+			map.descriptionText = cmd.toString()
+	}
+
+	results << createEvent(map)
```

Comment:
Since there is only 1 event in the "results" - You may also want to simply return createEvent(map)